### PR TITLE
fix #5314 Email Address Friendly Name Error

### LINKED ISSUE #5314: [SendMail] Email Address Friendly Name Error
### 📑 I have found these related issues/pull requests

NA

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

When selecting the SendGrid alert time and entering the "From Email" in the: name <email address> format, this gives a validation error:

image

If you click on the Test button, you get the email in the expected format, suggesting that this way of specifying the From Email is supported.

### 👟 Reproduction steps

On the Settings > Notifications tab, add a new notification type and select the SendGrid option.
Enter the email address such as: Uptime Kuma <test@test.com>
Select Test, if it's a valid email it will send.
Select the Save button, the above field validation error will show.

### 👀 Expected behavior

I would expect this option to save without error.

### 😓 Actual Behavior

Validation error: A part followed by '@' should not contain the symbol ' '.

### 🐻 Uptime-Kuma Version

2.0.0-beta.0-nightly-20241105100611

### 💻 Operating System and Arch

Windows 11

### 🌐 Browser

Chrome 130.0.6723.92 (Official Build) (64-bit)

### 🖥️ Deployment Environment

- Runtime: docker
- Database: embedded MariaDB
- Filesystem used to store the database on: 
- number of monitors: 

### 📝 Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by ABHIGYAN-MOHANTA**
# Description
Fixed: When selecting the SendGrid alert time and entering the "From Email" in the: name format, this gives a validation error

Fixes #5314 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- User interface (UI)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [x] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

https://github.com/user-attachments/assets/bc8478ee-eb90-4641-92f6-91fc9e28fe3b

### CODE REVIEW

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Looks reasonable.
I have left a few comments below (most of them try to make this part of the frontend less special)

**Comment by louislam** [comment]
In my opinion, just make it as a normal text field without regex checking, let "Test" button do the checking.

Because the email format regex usually will not be that simple, it may cause unpredictable bugs.

RFC 5322 / RFC 822


If really want to check it, finding a well-known library to verify the email address will be better.

**Comment by ABHIGYAN-MOHANTA** [comment]
I agree, even if we use a good email validation library like [Validator JS](https://www.npmjs.com/package/validator), we will still need to use regex for friendly name format validation, and if we are not going in the regex route, better to get rid of it. Sorry for introducing the complexity, I will just change the type to text. Thanks! Let's keep it straightforward, but if you decide otherwise, I am happy to implement validator.js as well! Thanks again!

**Comment by ABHIGYAN-MOHANTA** [comment]
@louislam, I have just kept it as a normal text field like you mentioned. Also, I realised, I can't really make ui changes just for SendGrid, it has to be for all providers, sorry about that 😅

File: src/components/notifications/SendGrid.vue [hunk 1]
```diff
@@ -38,10 +80,61 @@ export default {
...
     mounted() {
         if (typeof this.$parent.notification.sendgridSubject === "undefined") {
             this.$parent.notification.sendgridSubject = "Notification from Your Uptime Kuma";
         }
     },
+
+    methods: {
+        isValidEmailFormat(value) {
+            const friendlyEmailRegex = /^[^<>]+ <[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}>$/;
+            const emailOnlyRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
```
[the remark this answers is not in the document]
Comment (ABHIGYAN-MOHANTA) [reply src/components/notifications/SendGrid.vue]: Based on many github discussions, I dont think SMTPUTF8 is supported by Sendgrid.
[One such github discussion](https://github.com/Azure/azure-webjobs-sdk-extensions/issues/471#issuecomment-580164117)

**Pull request merged**